src/TextForEvent.js

```diff
@@ -248,6 +248,26 @@ function textForPowerEvent(event) {
     });
 }
 
+function textForWidgetEvent(event) {
+    const senderName = event.sender ? event.sender.name : event.getSender();
+    const previousContent = event.getPrevContent() ? event.getPrevContent() : {};
```

Comment:
I don't know how costly a called to `getPrevContent()` is, but in JavaScript I think the idiom is:

```
const previousContent = event.getPrevContent() || {};
```